api/v1_create_reward_code.go

```diff
@@ -90,17 +97,74 @@ func (app *ApiServer) v1CreateRewardCode(c *fiber.Ctx) error {
 		return err
 	}
 
-	// Verify the signature against authorized keys from config
-	matchedKey, err := verifySignatureAgainstKeys(req.Signature, config.Cfg.RewardCodeAuthorizedKeys)
+	_, err := verifySignatureAgainstKeys(req.Signature, config.Cfg.RewardCodeAuthorizedKeys)
 	if err != nil {
 		return fiber.NewError(fiber.StatusForbidden, "Unauthorized: "+err.Error())
 	}
 
+	// Generate a code
 	code, err := generateCode()
 	if err != nil {
 		return fiber.NewError(fiber.StatusInternalServerError, "Failed to generate code: "+err.Error())
 	}
 
+	var rewardAddress string
+
+	// Only create reward pool if deterministic secret is configured
+	if config.Cfg.LaunchpadDeterministicSecret != "" {
```

Comment:
Might be helpful to pull this stuff out into a util. I'm also going to need it in redemption.